File: packages/case-definition-dsl/src/matchers/http/HttpResponse.ts
import { HTTP_RESPONSE_MATCHER_TYPE } from '@contract-case/case-core-plugin-http-dsl';
import { AnyMatcher } from '../base';
import { AnyMatcherOrData } from '../../types';

/**
 * @public
 */
export interface HttpResponseExample {
  /* Any test-equivalence matcher or string or number for the HTTP status code accepted by this example (Recommended: `HttpStatusCode`) */
  readonly status: AnyMatcherOrData;

  /** What unique name, if any, to give to this response */
  readonly uniqueName?: string;
  /**
   * A Map of header names and associated test-equivalence matcher values
   * accepted by this example. If not provided, no header matching is performed */
  readonly headers?: AnyMatcherOrData;

  /**
   * A test equivalence matcher or json object that describes the body for this response. If not provided, no body matching is performed.
   */
  readonly body?: AnyMatcherOrData;
}

/**
 * Matches any HTTP Response with the provided properties
 * @public
 */
export class HttpResponse extends AnyMatcher {
  /** @internal */
  readonly '_case:matcher:type': typeof HTTP_RESPONSE_MATCHER_TYPE;

  readonly uniqueName?: string;

  readonly status: AnyMatcherOrData;

  readonly headers?: AnyMatcherOrData;

  readonly body?: AnyMatcherOrData;

  /**
   * Matches any HTTP Response with the provided properties
   * @param responseExample - An HttpResponseExample that describes the response
   */
  constructor(responseExample: HttpResponseExample) {
    super(HTTP_RESPONSE_MATCHER_TYPE);
    this.status = responseExample.status;
    if (responseExample.uniqueName !== undefined) {
      this.uniqueName = responseExample.uniqueName;
    }
    if (responseExample.headers !== undefined) {
      this.headers = responseExample.headers;
    }
    if (responseExample.body !== undefined) {
      this.body = responseExample.body;
    }
  }

  /**
   * For non-TypeScript implementations (see `AnyMatcher.toJSON`)
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/http/HttpResponse.java
package io.contract_testing.contractcase.dsl.matchers.http;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import jakarta.annotation.Generated;
import java.lang.Object;
import java.lang.String;
import javax.annotation.Nullable;
import javax.annotation.Nullable;
import javax.annotation.Nullable;
import javax.annotation.Nullable;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;
import lombok.Getter;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * Matches an HTTP response.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class HttpResponse implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * The HTTP status code accepted by this example (Recommended: Use the HttpStatusCode matcher)
   */
  @Getter
  @JsonProperty("status")
  private final Object status;

  /**
   * A dictionary object of header names and associated test-equivalence matcher values
   * accepted by this example.<p>If not provided, no header matching is performed
   */
  @Nullable
  @JsonInclude(Include.NON_NULL)
  @Getter
  @JsonProperty("headers")
  private final Object headers;

  /**
   * A dictionary object that describes the body for this response. If not provided, no body matching is performed.
   */
  @Nullable
  @JsonInclude(Include.NON_NULL)
  @Getter
  @JsonProperty("body")
  private final Object body;

  @Builder
  public HttpResponse(
    @NotNull final Object status,
    @Nullable final Object headers,
    @Nullable final Object body
  ) {
    this.type = "_case:HttpResponseMatcher";
    this.status = status;
    this.headers = headers;
    this.body = body;
  }
}
